### ISSUE TITLE
feature: Let users manually choose a key to focus on

### ISSUE THREAD

**Issue by aradzie** [aradzie/keybr.com#11]

**Comment by dsinkey** [comment]
I'd love this feature.  It seems like the algorithm for how the keys are chosen is based on the highest speed for that key. In my case the average speed for 'B' and 'J' is very low compared to the rest of the keys, however their top speed is much higher. This prevents me from being able to practice them, and get the average speed in line with the rest of the keys.

**Comment by dsinkey** [comment]
I've notice that there's been several changes and improvements to the UI lately.  I like the addition of the accuracy on the practice page, my accuracy percentage has already increased by 2 percent now that I'm focusing on it. I'm curious how long it will take to get this feature of being able to manually focus on your letter of choice.  This seems like it could be relatively easy to add.

The way I see it working is that the slider, which would be great to have as a text input to enter in the exact speed, would control the letters that are included in the practice, but if the letter you want to focus on is below that threshold it's adding to the letters included and that becomes the focus letter.

I use this app everyday and love it! Thanks for maintaining it!

**Comment by aradzie** [comment]
I've got a working prototype

![image](https://github.com/aradzie/keybr.com/assets/44386/69b12378-21cd-408d-8c63-abf059acd34b)

I'll try to finish it ASAP.

**Comment by adazmy** [comment]
Yes! often the whole progression is impacted by a weakness on one or two spots that is not related to speed but to finger used or bad habit. It would be great to have a way to manually focus the practice on a key!

**Comment by 100percentfast** [comment]
bumping this thread for exposure. was this ever implemented?

**Comment by mikerkelly** [comment]
👍 I think this would be a great improvement!

Taking it even further: it would be nice to be able to unlock all keys, but choose a set of keys that are eligible to be the focus of a lesson, and pick the weakest among those automatically. 

Rationale: once you reach a "reasonable" speed (50 WPM?) with all keys, the return-on-time-spent of focusing on your weakest key from the whole alphabet seems reduced, because those tend to be the rarest keys, which in English are much rarer than other keys. But that is what you practice if you set a target speed such that all keys are unlocked. So I have to choose between raising the target speed and not practising with all keys, or including all keys and only focusing on the weakest keys, which are usually very rare and don't much improve my average WPM. It is also kind of awkward and unpleasant to be constantly typing weird words rather than a mix of common, uncommon, and rare.

Z is always going to be among my weakest keys due to my keyboard design. And in British English it is very, very rare. So I don't get that much benefit from focusing many drills on that key. I want to focus on keys that are around my 5th weakest, say, while not neglecting my weakest keys altogether.

My 5th/6th weakest keys are M and W, which according to [wiki](https://en.wikipedia.org/wiki/Letter_frequency) are more than 30 times more common than Z, more so in British English. I would like to focus my lessons on letters like these without turning off my weakest keys altogether.

**Comment by prashantxNeo** [comment]
**Please add this feature:**
my average speed is **75wpm+** but letter **"Q"** speed is only 45wpm, **"Q" is kind of an exception** as it is **rarely** used but there are other keys like **{ C, B, V, X, }** which falls in range of **50-60wpm** which is bad so if i set goal less than 45wpm then i won't get individual keys to **practice** and if i set wpm at 50 i will be getting only "Q" lessons **endlessly** making **"Keybr"**'  **un-usable.**
